app/Models/MapMarker.php

```diff
@@ -141,6 +156,52 @@ public function isPolygon(): bool
         return $this->shape_id == MapMarker::SHAPE_POLY && !empty($this->custom_shape);
     }
 
+    /**
+     * Determine the type of the marker
+     * @return string
+     */
+    public function typeLabel(): string
+    {
+        if ($this->isPolygon()) {
+            return __('maps/markers.tabs.polygon');
+        } elseif ($this->isLabel()) {
+            return __('maps/markers.tabs.label');
+        } elseif ($this->isCircle()) {
+            return __('maps/markers.tabs.circle');
+        }
+        return __('maps/markers.tabs.marker');
+    }
+
+    /**
+     * Determine the icon of the marker
+     * @return string
+     */
+    public function datagridMarkerIcon(): string
+    {
+        if ($this->icon == 5 || $this->icon == 4) {
```

Comment:
Showing a pin icon for circle and polygons
![image](https://user-images.githubusercontent.com/3263440/197285247-b78fb500-c07e-4ac0-abcb-717d33492bbf.png)